qiskit/backends/local/statevector_simulator_cpp.py

```diff
@@ -48,7 +48,7 @@ def _run_job(self, qobj):
         # Add final snapshots to circuits
         for experiment in qobj.experiments:
             experiment.instructions.append(
-                QobjInstruction.from_dict({'name': 'snapshot', 'params': [final_state_key]})
+                QobjInstruction('snapshot', params=[final_state_key])
```

Comment:
Can you add the argument name when creating the `Qobj*` objects, as in:
```
QobjInstruction(name='snapshot', params=[final_state_key])
```

Even if it is not needed per the language, it feels safer to be explicit at least until we are sure the `Qobj` classes are in their "final" form as an extra precaution. If that sounds fine ... can you review all the instances in this PR?